go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -217,7 +216,17 @@ func (app *registryApplication) registerNode( // nolint: gocyclo
 
 	// Charge gas for node registration if signed by entity. For node-signed
 	// registrations, the gas charges are pre-paid by the entity.
-	isEntitySigned := sigNode.MultiSigned.IsSignedBy(newNode.EntityID)
+	//
+	// Yes, the detection logic is awful.  See VerifyRegisterNodeArgs,
```

Comment:
Since this check seems suboptimal, could we avoid doing it in case we are not in debug mode (e.g., `DebugAllowEntitySignedNodeRegistration`)? My understanding is that `VerifyRegisterNodeArgs` will fail in case entity signing is used and that debug-only flag is false?

Reply:
I don't think it's particularly performance intensive or anything, it's just awful from a design standpoint...

Reply:
Fixed anyway.